[rosbag2_transport] export shared_queues_vendor

## Description
The [current binary job](http://build.ros2.org/job/Fbin_uF64__rosbag2_bag_v2_plugins__ubuntu_focal_amd64__binary/7/consoleFull#console-section-15) for `rosbag2_bag_v2_plugins` is failing as it's using `rosbag2_transport` as a test dependency, yet `rosbag2_transport` is not exporting the `shared_queues_vendor` dependency which yields to a  

```
11:27:17 CMake Error at /opt/ros/foxy/share/ament_cmake_gmock/cmake/ament_add_gmock.cmake:69 (add_executable):
11:27:17   Target "test_rosbag2_play_rosbag_v2_end_to_end" links to target
11:27:17   "shared_queues_vendor::concurrentqueue" but the target was not found.
11:27:17   Perhaps a find_package() call is missing for an IMPORTED target, or an
11:27:17   ALIAS target is missing?
```

## Additional context
The solution here is trivial in a sense that `rosbag2_transport` needs to export `shared_vendor_queues` as a dependency.

For a quick workaround to get `rosbag2_bag_v2` released, I'll add the shared_queues dependency manually which then has to be removed in the first patch release.

--- a/rosbag2_transport/CMakeLists.txt
+++ b/rosbag2_transport/CMakeLists.txt
@@ -46,7 +46,6 @@ add_library(${PROJECT_NAME} SHARED
 target_include_directories(${PROJECT_NAME} PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>)
-
 ament_target_dependencies(${PROJECT_NAME}
   rcl
   rclcpp
@@ -85,7 +84,7 @@ install(
 ament_export_include_directories(include)
 ament_export_libraries(${PROJECT_NAME})
 ament_export_targets(export_${PROJECT_NAME})
-ament_export_dependencies(rosbag2_cpp rosbag2_compression yaml_cpp_vendor)
+ament_export_dependencies(rosbag2_cpp rosbag2_compression shared_queues_vendor yaml_cpp_vendor)
 
 function(create_tests_for_rmw_implementation)
   rosbag2_transport_add_gmock(test_info

--- a/rosbag2_transport/CMakeLists.txt
+++ b/rosbag2_transport/CMakeLists.txt
@@ -22,14 +22,12 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros REQUIRED)
-# TODO(Karsten1987) rosbag2_cpp has to be found first here
-# See https://github.com/ros2/ros2/issues/927
-find_package(rosbag2_cpp REQUIRED)
 find_package(rcl REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rcutils REQUIRED)
 find_package(rmw REQUIRED)
 find_package(rosbag2_compression REQUIRED)
+find_package(rosbag2_cpp REQUIRED)
 find_package(rmw_implementation_cmake REQUIRED)
 find_package(shared_queues_vendor REQUIRED)
 find_package(yaml_cpp_vendor REQUIRED)

